autogpt_platform/frontend/src/components/CustomNode.tsx

```diff
@@ -464,7 +465,7 @@ export function CustomNode({ data, id, width, height }: NodeProps<CustomNode>) {
 
     history.push({
       type: "ADD_NODE",
-      payload: { node: newNode },
+      payload: { node: { ...newNode, ...newNode.data } as CustomNodeData },
```

Comment:
We are intentionally flattening it out because history expects NodeData, which is a flattened out version of Node possible with additional values too